test/eigsolve.jl

```diff
@@ -359,3 +359,277 @@ end
     @test_logs realeigsolve(A, v, 1, :LM, Arnoldi(; tol=1e-12, verbosity=1))
     @test_logs (:info,) realeigsolve(A, v, 1, :LM, Arnoldi(; tol=1e-12, verbosity=2))
 end
+
+@testset "BlockLanczos - eigsolve for large sparse matrix and map input" begin
+    function toric_code_strings(m::Int, n::Int)
+        li = LinearIndices((m, n))
+        bottom(i, j) = li[mod1(i, m), mod1(j, n)] + m * n
```

Comment:
Maybe add a small comment here: there are `2 * m * n` qubits and you choose to enumerate all the horizontal ones from `1:m*n` and all the vertical ones as `(m*n+1):(2*m*n)`.